keras_nlp/metrics/rouge_l.py

```diff
@@ -0,0 +1,242 @@
+                f'"f1_score". Received: metric_type={metric_type}'
+            )
+
+        self.alpha = alpha
+        self.metric_type = metric_type
+        self.mask_token_ids = mask_token_ids
+
+        self._rouge_l_score = self.add_weight(
+            name="rouge_l_score",
+            initializer="zeros",
+            dtype=self.dtype,
+        )
+        self._number_of_samples = self.add_weight(
+            name="number_of_samples", initializer="zeros", dtype=self.dtype
+        )
+
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        # Both y_true and y_pred have shape: [batch_size, seq_len]. Note that
+        # they can also be ragged tensors with shape [num_samples, (seq_len)].
+
+        # If the input tensors are not ragged tensors, convert them to ragged
+        # tensors. `tf_text.metrics.rouge_l` expects ragged tensors.
+        if not isinstance(y_true, tf.RaggedTensor):
+            y_true = tf.RaggedTensor.from_tensor(y_true)
+        if not isinstance(y_pred, tf.RaggedTensor):
+            y_pred = tf.RaggedTensor.from_tensor(y_pred)
+
+        batch_size = tf.cast(y_true.nrows(), self.dtype)
+
+        y_true_mask = tf.cast(tf.ones_like(y_true), tf.bool)
+        y_pred_mask = tf.cast(tf.ones_like(y_pred), tf.bool)
+
+        if self.mask_token_ids is not None:
+            for mask_token_id in self.mask_token_ids:
+                y_true_mask = tf.logical_and(
+                    y_true_mask,
+                    tf.math.logical_not(tf.equal(y_true, mask_token_id)),
+                )
+                y_pred_mask = tf.logical_and(
+                    y_pred_mask,
+                    tf.math.logical_not(tf.equal(y_pred, mask_token_id)),
+                )
+
+            # Apply mask to both tensors.
+            y_true = tf.ragged.boolean_mask(y_true, y_true_mask)
+            y_pred = tf.ragged.boolean_mask(y_pred, y_pred_mask)
+
+        f1_scores, precisions, recalls = rouge_l(
+            y_true, y_pred, alpha=self.alpha
+        )
+        if self.metric_type == "precision":
+            scores = precisions
+        elif self.metric_type == "recall":
+            scores = recalls
+        else:
+            scores = f1_scores
+        self._rouge_l_score.assign_add(tf.reduce_sum(scores))
+        self._number_of_samples.assign_add(batch_size)
+
+    def result(self):
+        if self._number_of_samples == 0:
+            return 0.0
+        rouge_l_score = self._rouge_l_score / self._number_of_samples
+        return rouge_l_score
+
+    def reset_state(self):
+        self._rouge_l_score.assign(0.0)
+        self._number_of_samples.assign(0.0)
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "alpha": self.alpha,
+                "metric_type": self.metric_type,
+                "mask_token_ids": self.mask_token_ids,
+            }
+        )
+        return config
+
+
+def rouge_l(y_true, y_pred, alpha=0.5):
```

Comment:
We do not actually need to make this a standalone util. Just write 
```
f1_scores, precisions, recalls = tf_text.metrics.rouge_l(
        y_pred, y_true, alpha=alpha
    )
```
in the RougeL class.

Reply:
Thanks! My guess is there was some need to directly call the metrics computing function, which I am not sure if still applies. Let's keep it simple for now, and we can add the util if we find it is necessary.

Reply:
Great! Let me know if further changes are required